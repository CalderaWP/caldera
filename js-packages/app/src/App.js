import React from 'react';
import { Forms } from './features/forms/Forms';
import {ContactForm} from './forms/ContactForm';
import {TopBar} from './components/TopBar';
import AppBody from './components/AppBody';





export class App extends React.Component {

	state = {
		formsLoaded:false,
		forms : {

		},
		route: '/'
	};
	componentDidMount(){
		fetch('/wp-json/caldera-api/v1/forms' ).then(r => r.json() )
			.then( forms => this.setState({forms}))
			.catch( e => console.log(e))
			.then(() => this.setState({formsLoaded:true}));
	}

	setRoute(route){
		this.setState({route});
	}
	render() {
		const {
			forms,
			formsLoaded,
			route
		} = this.state;

		return (
			<div>
				<TopBar
					onChangeActive={(route) => this.setRoute(route)}
				/>
				<AppBody route={route} forms={forms}/>
			</div>
		);
	}
}
