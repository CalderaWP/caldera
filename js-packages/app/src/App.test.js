import renderer from 'react-test-renderer';
import React from 'react';
import { App } from './App';

//import "@babel/polyfill";
describe('App component', () => {
	it('Matches snapshot', () => {
		expect(renderer.create(<App />).toJSON()).toMatchSnapshot();
	});
});
