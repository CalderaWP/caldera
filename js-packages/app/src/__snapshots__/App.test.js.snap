// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Matches snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2196f3",
        "color": "#ffffff",
        "display": "flex",
        "flexDirection": "row",
        "height": 54,
        "padding": 8,
      }
    }
  >
    <h1
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "normal",
        }
      }
    >
      Caldera Pro
    </h1>
  </div>
  <div
    style={
      Object {
        "padding": 8,
      }
    }
  >
    Home Page
  </div>
</div>
`;
